examples/docker/Dockerfile

```diff
@@ -1,18 +1,38 @@
-FROM node:16-alpine
+FROM node:16.20.2-bookworm-slim as build
```

Comment:
I am a bit puzzled by this very specific version being used. What's the thoughts behind this? I guess the "slim" part means it's a streamlined image, which is good. If `16.20.2` is the latest `16` of that image, I think "node:16-bookworm-slim" would resolve to the same image, and leave the door open to eventually pick-up an updated `16.20.3` if there's ever one, without us having to update it manually.

Reply:
@frallax could you please answer @marcdumais-work questions?

Reply:
I am not a docker expert. The slim part is as you say (also if this is not so important for the build stage). Regarding the specific version, I just prefer being precise with versions to make things as reproducible as possible.

But the real reason for using this specific image is that I saw a similar project using this image and I tried that. My final goal was to shrink the size of the final image, and by using this multistage approach we do that, compared to what was on master before. I am sure there is room for improvement (both in term of final image size and reproducibility), and if you have any proposal I would be happy to review and test.

Reply:
I think using multi-stage build makes a lot of sense - nice improvement.